forth.asm

```diff
@@ -203,6 +203,28 @@ dict_entry tonumber, '>NUMBER', 0x0 ;( str -- n)
     push    ebx
     jmp     next
 
+interpret_if:
+    ; if stack nozero, advance instruction pointer
+    pop     eax
+    test    eax, eax
+    jnz     .true
+    add     esi, 0x8
+    jmp     next
+.true:
+    jmp     next
+
+
+dict_entry interpret, 'INTERPRET-WORD', 0x0 ; ( str -- )
+    call    docolon
+    dd      blank
+    dd      cword
+    dd      find
+    dd      interpret_if
```

Comment:
What if the number added to PC was not hard-coded, perhaps it might be the next cell in the instruction stream?  Maybe you could call this word something like `?BRANCH` or `BRANCHNZ`.